### ISSUE TITLE
Node::Simple mishandles child visibility of open `details` elements

### ISSUE THREAD

**Issue by aalong-tr** [teamcapybara/capybara#2242]
In Rails view specs, non-`summary` children of a `details` element are treated as not visible, even when the `details` element has the `open` attribute. This appears to be due to the xpath used at https://github.com/teamcapybara/capybara/blob/master/lib/capybara/node/simple.rb#L201, which does not respect the `open` attribute. Adding an attribute check to that clause of the xpath appears to resolve the issue:

```ruby
      (~x.self(:summary) & XPath.parent(:details)[!XPath.attr(:open)])
```

## Meta
Capybara Version: 3.28.0

Driver Information (and browser if relevant): N/A; 

## Expected Behavior
Visible children of a `details` element with the `open` attribute are treated as visible

## Actual Behavior
Non-`summary` children of a `details` element are always treated as not visible, irrespective of the `open` attribute. 

## Steps to reproduce
```ruby
require "capybara"
 
html = DATA.read
 
node = Capybara::Node::Simple.new(html)

puts node.has_selector?(".foo")

__END__
 
<!doctype html>
<html>
  <body>
    <details open>
      <div class='foo' />
    </details>
  </body>
</html>
```

**Comment by twalpole** [comment]
Thanks for mentioning this, I’ll fix it this weekend

**Comment by twalpole** [comment]
You can try the `simple_details` branch -- Assuming all tests pass it will get merged into master this weekend.

**Comment by aalong-tr** [comment]
Thank you!

**Comment by twalpole** [comment]
This is in master now, along with the ability to open/close details elements by clicking on the element when using the rack_test driver.

**Comment by twalpole** [comment]
This is released in 3.29.0

**Comment by aalong-tr** [comment]
Thanks again for the quick turnaround, I appreciate the help!
